extensions/amp-vine/0.1/amp-vine.js

```diff
@@ -0,0 +1,72 @@
+
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+
+class AmpVine extends AMP.BaseElement {
+  /** @override */
+  preconnectCallback(onLayout) {
+    // the Vine iframe
+    this.preconnect.url('https://vine.co');
```

Comment:
@dvoytenko would it be appropriate to pass in the the `opt_alsoConnecting` flag here like so:

```js
this.preconnect.url('https://www.youtube.com', onLayout);
```